CRM/MembershipExtras/BAO/MembershipPeriod.php

```diff
@@ -270,16 +298,32 @@ private static function doesOverlapWithOtherActivePeriods($periodParams) {
 
   private static function updateMembershipDates($membershipPeriod) {
     $membershipId = $membershipPeriod->membership_id;
-    $joinDate = $startDate = self::getFirstActivePeriod($membershipId)['start_date'];
-    $endDate = self::getLastActivePeriod($membershipId)['end_date'];
+    $firstActivePeriod = self::getFirstActivePeriod($membershipId);
+    $lastActivePeriod = self::getLastActivePeriod($membershipId);
 
-    civicrm_api3('Membership', 'create', [
+    $joinDate = $startDate = CRM_Utils_Array::value('start_date', $firstActivePeriod, '');
+    $endDate = CRM_Utils_Array::value('end_date', $lastActivePeriod, '');
+
+    $params = [
       'id' => $membershipId,
-      'join_date' => $joinDate,
-      'start_date' => $startDate,
-      'end_date' => $endDate,
       'skipStatusCal' => 0,
-    ]);
+    ];
+
+    if (!empty($joinDate)) {
+      $params['join_date'] = $joinDate;
+    }
+
+    if (!empty($startDate)) {
```

Comment:
or remove this and change the above to : 

```
if (!empty($joinDate)) {
      $params['join_date'] = $joinDate;
      $params['start_date'] = $joinDate;
}
``` 

you can also remove $startDate variable from line 304 and line 324